### DISCUSSION TITLE
`threads` option is overridden before passing it to `number-processes` arg

### DISCUSSION THREAD

**Discussion by david-mart** [osm-search/Nominatim#3218] [Q&A] [answered]
Hello, first of all I want to thank you for this amazing library.

When running `ps aux` during the import stage on a docker container
```
python3 /usr/local/bin/nominatim import --osm-file /nominatim/data.pbf --threads 24 --no-updates
/usr/local/lib/nominatim/osm2pgsql --hstore --latlon --slim --log-progress true --number-processes 1 --cache 0 --style /nominatim/style.lua --output flex --create --flat-nodes /nominatim/flatnode/flatnode.file /nominatim/data.pbf
``` 

I noticed that `--number-processes` is set to 1. After briefly looking through the code, I see that this value is [overridden](https://github.com/osm-search/Nominatim/blob/54cb9a33b1f62b91ce7b3dd9e4773c46cc04c2b8/nominatim/tools/database_import.py#L98C9-L98C9) in the import script.

I was wondering if this is intentional, and if it's possible to bump it up?

**Comment by mtmail** [comment, marked as answer]
Quoting from https://github.com/osm-search/Nominatim/pull/1408#issuecomment-507817174 (@lonvia is one of the maintainers of osm2psql and maintainer of Nominatim)

The number of processes was set to 1 intentionally. The osm2pgsql import itself is not multithreaded (or rather it is but not in any way that is influenced by this parameter). The only phase that is parallel is indexing. In this phase a high number of threads tends to be counter-productive. In my experience it is better to increase the maintenance memory to make creating a single index faster than increasing the number of parallel threads.
